BetonQuest-core/src/main/java/pl/betoncraft/betonquest/compatibility/citizens/CitizensVariable.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * BetonQuest - advanced quests for Bukkit
+ * Copyright (C) 2016  Jakub "Co0sh" Sapalski
+ * 
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package pl.betoncraft.betonquest.compatibility.citizens;
+
+import net.citizensnpcs.api.CitizensAPI;
+import net.citizensnpcs.api.npc.NPC;
+import org.bukkit.Location;
+import pl.betoncraft.betonquest.Instruction;
+import pl.betoncraft.betonquest.InstructionParseException;
+import pl.betoncraft.betonquest.api.Variable;
+
+/**
+ * Provides information about a citizen npc.
+ *
+ * Format:
+ *   %citizen.<id>.<key>%
+ *
+ * Keys:
+ *   * name - (default) Return citizen name
+ *   * location  - Return citizen location. x;y;z;world;yaw;pitch
+ */
+public class CitizensVariable extends Variable {
+
+	private int id;
+	private String key;
+
+	public CitizensVariable(Instruction instruction) throws InstructionParseException {
+		super(instruction);
+
+		String [] parts = instruction.getInstruction().split("\\.");
```

Comment:
This could be simplified by using [`instruction.getPositive()`](https://betonquest.pl/javadocs/pl/betoncraft/betonquest/Instruction.html#getPositive--) and [`instruction.next()`](https://betonquest.pl/javadocs/pl/betoncraft/betonquest/Instruction.html#next--), as the instruction object has some inbuild logic that handles parsing of arguments and throwing errors if missing/invalid.

But your self-made logic is completly fine to. :+1:

Reply:
Changed to remain consistent (and it means 1 place to fix future bugs).